### PR TITLE
Issue296 zindexanslayer test

### LINKED ISSUE #296: ModelIndex::zIndexAndLayer is not tested
It seemed like such a simple function, but when messing around with array index ordering, it becomes a headache.

Write a test that ensures that `zIndexAndLayer` accesses the correct point in a three-dimensional `ModelArray`.

### PR DESCRIPTION
**Pull request by timspainNERSC**
Add a test for the ModelArray::zIndexAndLayer function.

Will close #296 
 
> It seemed like such a simple function, but when messing around with array index ordering, it becomes a headache.
> 
> Write a test that ensures that zIndexAndLayer accesses the correct point in a three-dimensional ModelArray.

### CODE REVIEW

File: core/test/ModelArray_test.cpp [hunk 1]
```diff
@@ -255,6 +255,36 @@ TEST_CASE("Location from index")
     REQUIRE(loc[1] == y);
     REQUIRE(loc[2] == z);
 }
+
+// Test the zIndexAndLayer function
```
Comment (TomMelt) [inline core/test/ModelArray_test.cpp:259]: I think you should expand the description a little bit. At least something similar to issue #296

File: core/test/testmodelarraydetails/ModelArrayDetails.cpp [hunk 2]
```diff
@@ -43,6 +49,7 @@ const std::map<ModelArray::Type, std::string> ModelArray::typeNames = {
     { ModelArray::Type::ONED, "OneDField" },
     { ModelArray::Type::TWOD, "TwoDField" },
     { ModelArray::Type::DOSD, "TwoDField" },
+    { ModelArray::Type::DOSD, "ThreeDField" },
```
Comment (TomMelt) [inline core/test/testmodelarraydetails/ModelArrayDetails.cpp:52]: I think this should be `ModelArray::Type::THREED`

As a side note, do we explain anywhere what `DOSD` is? how does it differ from TWOD. I am guessing `DOS` is the number 2 in Spanish?

**Review by TomMelt — CHANGES_REQUESTED** [review]

File: core/test/ModelArray_test.cpp (hunk 1 above)
Comment (timspainNERSC) [reply core/test/ModelArray_test.cpp]: Copied from the #296 description (with minor editing).

File: core/test/testmodelarraydetails/ModelArrayDetails.cpp (hunk 2 above)
Comment (timspainNERSC) [reply core/test/testmodelarraydetails/ModelArrayDetails.cpp]: > I think this should be ModelArray::Type::THREED

Indeed it should.

DOSD has now been renamed to the more descriptive ZUFIELD.

**Review by TomMelt — APPROVED** [review]
Looks good. Thanks for making the changes.

**Pull request merged**